File: Rubik.HTML/Server/Schema/LevelInfo.cs
using System;
using System.Web;
using System.ComponentModel;
using System.Collections.Generic;

namespace Rubik.Server.Schema
{
    public class LevelInfo : InfoObject
    {
        public HierarchyInfo Hierarchy
        {
            get; set;
        }

        public DimensionInfo Dimension
        {
            get; set;
        }

        public CubeInfo Cube
        {
            get; set;
        }

        public List<MemberInfo> Members
        {
            get; set;
        }
    }
}

File: Rubik.HTML/Lib/DataHub/Schema/LevelInfo.ts
module Rubik.DataHub {
    export class LevelInfo extends InfoObject {
        get ObjectType(): ObjectTypeEnum { return ObjectTypeEnum.Level; }

        Cube_UniqueName: string;
        Dimension_UniqueName: string;
        Hierarchy_UniqueName: string;        

        get Hierarchy(): Promise<HierarchyInfo> {
            return this.factory.GetAncestorInfo(Rubik.DataHub.ObjectTypeEnum.Hierarchy, this) as Promise<HierarchyInfo>;
        }
        get Dimension(): Promise<DimensionInfo> {
            return this.factory.GetAncestorInfo(Rubik.DataHub.ObjectTypeEnum.Dimension, this) as Promise<DimensionInfo>;
        }
        get Cube(): Promise<CubeInfo> {
            return this.factory.GetAncestorInfo(Rubik.DataHub.ObjectTypeEnum.Cube, this) as Promise<CubeInfo>;                    
        }
        get Members(): Promise<MemberInfo[]> {
            return this.factory.GetInfoObjectCollection(Rubik.DataHub.ObjectTypeEnum.Member, this) as Promise<MemberInfo[]>; 
        }
    }
}